src/testdir/test_window_cmd.vim

```diff
@@ -1557,4 +1557,47 @@ func Test_window_alloc_failure()
   tabonly
 endfunc
 
+func Test_splitscroll_with_splits()
+  set splitbelow
+  set scrolloff=0
+  call setline(1, range(1,256))
+  norm j
+  split
+  wincmd p
+  call assert_equal(2, line("w0"))
+  quit
+
+  set splitscroll=1
+  norm ggj
+  split
+  wincmd p
+  call assert_equal(1, line("w0"))
+  quit
+
+  set scrolloff=5
+  split
+  wincmd p
+  norm ggL
+  let line = line(".") + 1
+  quit
+  norm gg
+  execute 'norm' l:line . 'G'
+  split
+  wincmd p
+  call assert_equal(2, line("w0"))
+  quit
+
+  set scrolloff=0
+  set splitscroll=2
+  norm ggL
+  split
+  quit
+  call assert_equal(winheight(0), line("."))
+  split
+  wincmd p
+  call assert_equal(winheight(0), line("."))
```

Comment:
I made the option non-boolean in the latest commit as per https://github.com/neovim/neovim/pull/19243#issuecomment-1179607771. I also added some tests but I'm running into an issue.

I cant get this test to pass even though manually running the commands yields the expected results. I expect this has something to do with https://github.com/vim/vim/blob/34be31ebaa1ed3f139928d0f7a1a6d1993621e31/src/window.c#L4943-L4945
Where we move the cursor to the last visible line when moving back to the original window in the case where `'splitscroll' == 2` and the cursor was behind the newly opened split. I suspect this is not the correct way to set the cursor position and is what causes the assertion to fail when run inside a VimL function.

I suspected some function from `move.c` should be called to update/invalidate but I didn't find it yet. What would be the correct way to update the cursor, or am I on the wrong track?

Reply:
I figured out `w_botline` is not the expected value here when running the test, hence the cursor is not set and the test fails. Is `w_botline` being outdated expected when running VimL functions?

Exchanging `w_botline` for `w_topline + w_height` resolves the issue and makes the test pass. Would this be a reasonable workaround or should I be calling some sort of function from `move.c` to update `w_botline`?